### ISSUE TITLE
-Wshadow does not warn for `static` variables

### ISSUE THREAD

**Issue by alexkaratarakis** [llvm/llvm-project#62850]
Min repro:

```C++
// Compile with -Wshadow
int main()
{
    static bool CONDITION = false;

    {
        static bool CONDITION = true;
    }
}
```

expected results: `declaration of 'CONDITION' shadows a previous local [-Wshadow]`
actual results: No warnings

gcc warns when the diagnostic is enabled, but clang does not.

Demo: https://godbolt.org/z/WjPTMvYP8

**Comment by hazohelet** [comment]
Proposed fix: https://reviews.llvm.org/D151214
